src/test/groovy/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPostRepositorySpec.groovy

```diff
@@ -39,24 +38,28 @@ class BlogPostRepositorySpec extends SpringContextAwareSpecification {
 
             blogPostRepository.save(blogPosts)
         when:
-            def latestPosts = blogPostRepository.findLatestPosts(new PageRequest(0, blogPosts.size()))
+            List<BlogPost> latestPosts = blogPostRepository.findLatestPosts(new PageRequest(0, blogPosts.size()))
         then:
+            // not moderated posts first ...
             latestPosts[0].isNotModerated()
             latestPosts[1].isNotModerated()
             latestPosts[0].getPublishedDate().isAfter(latestPosts[1].getPublishedDate())
 
-            latestPosts[2].isRejected()
-            latestPosts[3].isRejected()
-            latestPosts[2].getPublishedDate().isAfter(latestPosts[3].getPublishedDate())
+            // ... then moderated ones ordered by published date regardless of an approval
+            latestPosts[2].isModerated()
+            latestPosts[3].isModerated()
+            latestPosts[4].isModerated()
+            latestPosts[5].isModerated()
 
-            latestPosts[4].isApproved()
-            latestPosts[5].isApproved()
+            latestPosts[2].getPublishedDate().isAfter(latestPosts[3].getPublishedDate())
+            latestPosts[3].getPublishedDate().isAfter(latestPosts[4].getPublishedDate())
             latestPosts[4].getPublishedDate().isAfter(latestPosts[5].getPublishedDate())
     }
 
     private Blog aBlog() {
-        def blog = new Blog(jsonId: 1L, author: "Top Blogger", rss: "http://topblogger.pl/", dateAdded: LocalDateTime.now(), blogType: PERSONAL)
-        return blogRepository.save(blog)
+        return blogRepository.save(
+                new Blog(jsonId: 1L, author: "Top Blogger", rss: "http://topblogger.pl/",
```

Comment:
Try to avoid _GroovyBeans_ provided constructors and prefer `Blog.builder()` instead. _GroovyBeans_ style constructors are error prone because if the field name is changed then you will get feedback late at runtime. Using builder you will benefit from IDE code completion and errors will be highlighted instantaneously when test is broken by subject code changes.